tests/helpers/fake_redis_node.tcl

```diff
@@ -0,0 +1,58 @@
+# A fake Redis node for replaying predefined/expected traffic with a client.
+#
+# Usage: tclsh fake_redis_node.tcl PORT COMMAND REPLY [ COMMAND REPLY [ ... ] ]
+#
+# Commands are given as space-separated strings, e.g. "GET foo", and replies as
+# RESP-encoded replies minus the trailing \r\n, e.g. "+OK".
```

Comment:
@zuiderkwast i truly admire the resourcefulness, but i'm not sure we wanna go down that road.
mocking redis would only be useful to test redis tooling (such as redis-cli), and i think would have several disadvantages:
- if the mock is wrong, and behaves incorrectly (e.g. a typo such as AKS instead of ASK), or any problem with the protocol, the test will pass and we won't realize the code is bad (e.g. this PR fixes a character case issue)
- may go outdated if redis is changed and the tests will still pass.

I see two other possibilities if we wanna test it (p.s. i'm also ok with not testing it).
1. add some test suite infra (maybe in server.tcl) that let's us bring up a single redis server that's part of an imaginary cluster. i.e. create a server that behaves as a cluster node (thinks it's part of a cluster), but without actually creating a cluster.
2. i see the cluster tests (`./tests/cluster/*`) do interact with redis-cli, so i suppose we can add a cli-test in that folder using the framework to create clusters.

Reply:
@oranagra Thanks ;-)

How do you consistently provoke an ASK redirect in an "imaginary cluster" or a real cluster?

AFAICT, the reason the ASK bug is there is that there was no way to test it, so not testing it at all would definitely be sub optimal.

Of course the mocked Redis can be wrong. That's why it needs to be minimal and with as little logic as possible. We can test the mocked Redis by checking that it's consistent with real Redis for some regular commands. We can use redis-cli to do that in a test case. How about that?

Mocked Redis is used to test ASK redirects in client libs. It's the only way to test it that I've found actually. To consistently provoke it in real redis would require injecting some sleep/pause in the cluster migration.

Reply:
we have `DEBUG AOF-FLUSH-SLEEP` and `CONFIG SET key-load-delay`.
so maybe it's time to add something similar to be able to consistently debug ASK (may help other tests, not just redis-cli).

Reply:
> maybe it's time to add something similar to be able to consistently debug ASK (may help other tests, not just redis-cli)

There is no real test for Redis sending ASK redirects correctly. That can be added with such a mechanism.